### PR TITLE
Do not upcase State abbreviation

### PR DESCRIPTION
**Pull request by mllocs**
#### What? Why?

I propose to remove the constraint of having uppercased State abbreviations. This would allow us to have other than uppercased State abbreviations. 

screenshot 2017-11-16 15 48 11

⚠️  **It would show the same format as it is stored in the database. So, instances that want abbreviations in uppercase, should update the `abbr` column in the `spree_states` table.**

Also, as a note, the `upcase` ruby method doesn't play well with accents nor other symbols. 

#### What should we test?

Forms and sections that have addresses. 

#### Release notes

Remove constraint of uppercased State abbreviations.
**PLEASE MAKE SURE YOU DOUBLE CHECK HOW STATES APPEAR IN YOUR INSTANCE AFTER DEPLOY**

### CODE REVIEW

**Comment by oeoeaio** [comment]
I've approved because the code is fine. We still need to decide (through user testing) whether the change is something that we can manage as a community. Any testing notes very welcome.

**Comment by oeoeaio** [comment]
Also note @mllocs there is a failing feature spec that tests that state text is upcased: `spec/features/consumer/registration_spec.rb:52`.

**Comment by daniellemoorhead** [comment]
>We still need to decide (through user testing) whether the change is something that we can manage as a community. Any testing notes very welcome.

@oeoeaio reckon it would be worth having @sstead test this next Wednesday and provide her feedback about it? If yes, add it to the testing queue 😄

**Comment by mllocs** [comment]
@oeoeaio fixed the specs 💪

**Comment by daniellemoorhead** [comment]
@oeoeaio are we ready to have Sally test this? Worth staging for her today?

**Comment by daniellemoorhead** [comment]
New year testing for you @sstead, I added it to your list on the little whiteboard of things to test :)

**Comment by sstead** [comment]
Testing notes:

- The abreviations don't carry to the shops/producer directory (should they?)
- There's a translation issue in the customer's saved billing/shipping details form

https://docs.google.com/document/d/1hRF1Br5SUgBTSEUS9e8SXJ4_How7w-GS6oJKySXo9qo/edit?usp=sharing

**Comment by daniellemoorhead** [comment]
Back to you @mllocs - see @sstead testing notes above. Very sorry about the delay in doing this, let us know when it's ready to go again :)

**Comment by mllocs** [comment]
I personally thing that if a State name abbreviation only make sense in capitals, it should be written in capitals in the database. In that case, we show the state abbreviation as it is written in the database. Hm... what do you think?

**Comment by myriamboure** [comment]
Yes I would vote for that @mllocs so then it's the responsability of the super admin of course to write those abbreviation and make sure they "looks good" on the front end directories as well. So I would say let's fix that :-)

**Comment by sauloperez** [comment]
Agree. Let's move it forward :muscle:

**Comment by daniellemoorhead** [comment]
@sstead @oeoeaio is this proposed solution (using caps in the copy) ok for AU?

**Comment by sstead** [comment]
I'm sure this is also fine for Aus @daniellemoorhead .

**Comment by enricostano** [comment]
Mh, so... are we going to migrate all instances state data? :thinking:

**Comment by daniellemoorhead** [comment]
@enricostano 
giphy

:grin

**Comment by myriamboure** [comment]
@mllocs any move on this one?
- Still one thing to fix: "The abreviations don't carry to the shops/producer directory (should they?)"
- I think we all agree we are ok with what you proposed.
Confirm @enricostano or does the migration consequence is balancing your mind?

**Comment by enricostano** [comment]
Talked offline with @mllocs , let's move forward BUT I added some detail to the Release Note in the description in order to remind you that you may need to fix the states in your DB...

TL;DR if you are responsible for any instance please check how states look like in your DB

**Comment by enricostano** [comment]
> Still one thing to fix: "The abreviations don't carry to the shops/producer directory (should they?)"

OK, it seems that there is still one issue to fix. Back to the developer :grimacing: .

**Comment by myriamboure** [comment]
Hi @mllocs ! Any progress on that one? Let us know if anything remains unclear ;-)

**Comment by daniellemoorhead** [comment]
@mllocs @enricostano @sauloperez this has been sitting without action for a month now. Is there much remaining to do, can it be quickly finished, or should we perhaps close it for now and pick it up later when there is time?

**Comment by enricostano** [comment]
I guess we'll need to look for another developer to pick it up.

**Comment by luisramos0** [comment]
Hi guys,

I think I fixed this. I just removed the uppercase statement from the views on both producers and shops list in the frontoffice.
Please let me know if there's anything else that needs to be fixed on this story.

### Tech
I am just not sure what's the git strategy I should use to update this PR.
I dont have access to coopdevs/ofn where this branch is so I created a PR on it with my change:
https://github.com/coopdevs/openfoodnetwork/pull/1

Thanks,
Luís

**Comment by myriamboure** [comment]
Maybe @mkllnk or @sauloperez can tell you about how to modify original branch or add your commit to it.

**Comment by mkllnk** [comment]
@luisramos0 will provide another pull request to address the last point. But this pull request is already good to go. Moving it to test ready.

**Comment by daniellemoorhead** [comment]
@luisramos0 am assigning you to this PR just so that it has an owner just in case there are problems found when it is tested. Hopefully it will be perfect and can be merged without any changes 😄

**Comment by myriamboure** [comment]
There is no problem actually with the list of regions, see my comments @RachL in your testing notes and I added some screenshares ;-) I think this PR is ready to go then!
